src/libstd/sys/unix/time.rs

```diff
@@ -317,6 +318,20 @@ mod inner {
             t
         }
 
+        #[cfg(target_os = "dragonfly")]
+        pub fn now(clock: libc::c_int) -> Timespec {
+            let mut t = Timespec {
+                t: libc::timespec {
+                    tv_sec: 0,
+                    tv_nsec: 0,
+                }
+            };
+            cvt(unsafe {
+                libc::clock_gettime(clock as libc::c_ulong, &mut t.t)
+            }).unwrap();
+            t
+        }
```

Comment:
Could the duplication here be reduced from the whole function to just perhaps a typedef?